parlai/core/pytorch_data_teacher.py

```diff
@@ -326,7 +325,7 @@ def __getitem__(self, index):
                 return (index, ep)
 
     def __len__(self):
-        return self.num_eps
+        return int(self.num_episodes() * max(self.opt.get('num_epochs', 1), 1))
```

Comment:
what's this for?

Reply:
This is intended to fix the issue that a new data-loading iterator was constructed for each teacher in the `reset_data` function; now, only one iterator is created, and does not stop iterating until the appropriate amount of examples have been returned. I will change this so that it works for validation/test (as right now it looks like it would do too many iterations)